src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -274,5 +274,107 @@ public static void CopyTo<T>(this T[] array, Span<T> destination)
         {
             new ReadOnlySpan<T>(array).CopyTo(destination);
         }
+
+        /// <summary>
+        /// Determines whether two sequences overlap.
+        /// </summary>
+        public static unsafe bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second)
+        {
```

Comment:
Should this just be:
```C#
public static bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second) => first.Overlaps(second, out _);
```

Reply:
Any performance differences? If not, it should.

Speaking of performance, should the methods have AggressiveInlining?

Reply:
cc @jkotas 

> Any performance differences? If not, it should.

If there is a penalty, maybe refactor into an internal helper to avoid code duplication (between all 4 methods, since the real calculation is common between Span/RoS).

Reply:
it should call:
```
unsafe bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> second)
```
not the one with out parameter.

Reply:
This?

```csharp
public static bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second) => Overlaps((ReadOnlySpan<T>)first, second, out _);
public static bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second, out int elementOffset) => Overlaps((ReadOnlySpan<T>)first, second, out elementOffset);
public static bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> second) => Overlaps(first, second, out _);
public static bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> second, out int elementOffset) // full implementation
```

Reply:
I believe this is what @nietras is suggesting:

```C#
public static bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second) => Overlaps((ReadOnlySpan<T>)first, second);
public static bool Overlaps<T>(this Span<T> first, ReadOnlySpan<T> second, out int elementOffset) => Overlaps((ReadOnlySpan<T>)first, second, out elementOffset);
public static bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> second) => Overlaps(first, second, out _); // or full implementation since there will be a perf difference (doing the throw away elementOffset calculation)
public static bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> second, out int elementOffset) // full implementation
```

Reply:
I think that the code duplication is ok here.